solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -119,6 +126,10 @@ public DocCollection(String name, Map<String, Slice> slices, Map<String, Object>
     this.activeSlicesArr = activeSlices.values().toArray(new Slice[activeSlices.size()]);
     this.router = router;
     this.znode = ZkStateReader.getCollectionPath(name);
+    this.configName = String.valueOf(props.get(CONFIGNAME_PROP)) == null ? String.valueOf(props.get(COLLECTION_CONFIG_PROP)) : String.valueOf(props.get(CONFIGNAME_PROP));
```

Comment:
Pick one -- probably CONFIGNAME_PROP, not also COLLECTION_CONFIG_PROP.  I believe COLLECTION_CONFIG_PROP is only used at the HTTP API level, not internally.